core/frontend/src/components/vehiclesetup/overview/OnboardSensors.vue

```diff
@@ -0,0 +1,185 @@
+    },
+  },
+  mounted() {
+    mavlink.setMessageRefreshRate({ messageName: 'SCALED_PRESSURE', refreshRate: 1 })
+    mavlink.setMessageRefreshRate({ messageName: 'SCALED_PRESSURE2', refreshRate: 1 })
+    mavlink.setMessageRefreshRate({ messageName: 'VFR_HUD', refreshRate: 1 })
+  },
+  methods: {
+    imu_is_calibrated(imu: deviceId) {
+      const param_radix = imu.param.split('_ID')[0]
+      const offset_params_names = [`${param_radix}OFFS_X`, `${param_radix}OFFS_Y`, `${param_radix}OFFS_Z`]
+      const scale_params_names = [`${param_radix}SCAL_X`, `${param_radix}SCAL_Y`, `${param_radix}SCAL_Z`]
+      const offset_params = offset_params_names.map(
+        (name) => autopilot_data.parameters.filter((parameter) => parameter.name === name).first(),
+      )
+      const scale_params = scale_params_names.map(
+        (name) => autopilot_data.parameters.filter((parameter) => parameter.name === name).first(),
+      )
+      const is_at_default_offsets = offset_params.map((param) => param?.value).every((value) => value === 0.0)
+      const is_at_default_scale = scale_params.map((param) => param?.value).every((value) => value === 1.0)
+      return !is_at_default_offsets || !is_at_default_scale
+    },
+    is_water_baro(baro: deviceId) {
+      if (['MS5837', 'MS5611', 'KELLERLD'].includes(baro?.deviceName ?? '--')
+      && autopilot.vehicle_type === 'Submarine') {
+        return true
+      }
+      return false
+    },
+    compass_is_calibrated(imu: deviceId) {
+      const compass_number = imu.param.split('COMPASS_DEV_ID')[1]
+      const offset_params_names = [
+        `COMPASS_OFS${compass_number}_X`,
+        `COMPASS_OFS${compass_number}_Y`,
+        `COMPASS_OFS${compass_number}_Z`,
+      ]
+      const diagonal_params_names = [
+        `COMPASS_ODI${compass_number}_X`,
+        `COMPASS_ODI${compass_number}_Y`,
+        `COMPASS_ODI${compass_number}_Z`,
+      ]
+
+      const offset_params = offset_params_names.map(
+        (name) => autopilot_data.parameters.filter((parameter) => parameter.name === name).first(),
+      )
+      const diagonal_params = diagonal_params_names.map(
+        (name) => autopilot_data.parameters.filter((parameter) => parameter.name === name).first(),
+      )
+      if (offset_params.includes(undefined) || diagonal_params.includes(undefined)) {
+        notifier.pushError('PARAM_MISSING', 'Unable to find Compass parameters')
+      }
+
+      const scale_param_name = `COMPASS_SCALE${compass_number}`
+      const scale_param = autopilot_data.parameters.filter((parameter) => parameter.name === scale_param_name).first()
+      const is_at_default_offsets = offset_params.map((param) => param?.value).every((value) => value === 0.0)
+      const is_at_default_diagonals = diagonal_params.map((param) => param?.value).every((value) => value === 0.0)
+      return !is_at_default_offsets || !is_at_default_diagonals || scale_param?.value !== 0.0
```

Comment:
We need to check if the list is not empty... because...

![image](https://user-images.githubusercontent.com/1215497/214405735-20cbbcbb-0534-48d0-a24e-c14d21b329d7.png)